apps/verticalhome/js/app.js

```diff
@@ -150,6 +179,177 @@
 
     },
 
+    saveAllPinnedApps: function(){
+      var pinnedItems= app.pinnedAppsManager.items;
+      if(this.deletedApps.length>0){
+        for(var x = 0; x < pinnedItems.length; x++){
+          if(pinnedItems[x].element.style.display == 'none'){
+            pinnedItems.splice(x,1);
+          }
+        }
+
+        for(var z = 0; z < this.deletedApps.length; z++){
+          app.itemStore.deletePinnedAppItem(this.deletedApps[z]);
+        }
+      }
+
+        for(var i = 0; i<pinnedItems.length; i++){
+          var obj = {};
+
+          if(pinnedItems[i].entry!=null){
+
+              obj.name = pinnedItems[i].entry.name;
+              obj.manifestURL = typeof(
+                pinnedItems[i].entry.manifestURL
+                ) == 'undefined'?
+                pinnedItems[i].entry.targetApp.manifestURL
+                : pinnedItems[i].entry.manifestURL;
+
+              obj.icon = typeof(
+                pinnedItems[i].entry.icon
+                )=='undefined'?
+                pinnedItems[i].entry.icon.baseURI
+                :pinnedItems[i].entry.icon;
+
+              obj.index = pinnedItems[i].index;
+
+              app.itemStore.savePinnedAppItem(obj);
+
+          }
+        }
+
+       this.showNotif('Changes saved');
```

Comment:
will  this 'notification' be shown to user? If so, you shall use new localizable strings for such text.